### PR TITLE
Function in TU to build preprocessed tokens for consumer.

### PR DESCRIPTION
**Pull request by natsegal**
Also, implement multiple sources and targets as a dictionary with one 'main' source/target.

### CODE REVIEW

**Comment by natsegal** [comment]
@guillaumekln I'm still a little "fuzzy" about this approach, so feel free to criticize and comment.
Also, CodeFactor is complaining about "complex code", not sure how to fix it...

File: nmtwizard/preprocess/tu.py [hunk 1]
```diff
@@ -210,22 +220,34 @@ def __init__(self,
                  alignment=None,
                  source_tokenizer=None,
                  target_tokenizer=None):
-        self.__source = [TranslationSide(source, tokenizer=source_tokenizer)]
+        self.__source = {"main": TranslationSide(source, "source", tokenizer=source_tokenizer)}
```
Comment (guillaumekln) [inline nmtwizard/preprocess/tu.py:223]: Do we need a dictionary here? Can't we just consider the first index as "main" and others as "extra"?

Comment (natsegal) [reply nmtwizard/preprocess/tu.py:223]: In source we probably could. But in target, as we discussed, in inference we wouldn't know if the first target came from a translation prefix or it's an extra target. We could use a list with names if you prefer. I thought dictionary will ensure unique names.

**Review by guillaumekln — CHANGES_REQUESTED** [review]

File: nmtwizard/preprocess/tu.py [hunk 2]
```diff
@@ -302,41 +336,87 @@ def _invalidate_alignment(self):
...
+            self.__target[key].detok = detok
+            if key == "main":
                 self._invalidate_alignment()
 
+    def build_preprocessed_tokens(self, side):
+        tok = None
+        if side == "source":
+            tok = self.__source
+        else:
+            tok = self.__target
```
[the remark this answers is not in the document]
Comment (natsegal) [reply nmtwizard/preprocess/tu.py:389]: Yes, this source/target checks are everywhere and are error-prone. Maybe we should change it to enum everywhere later.

**Pull request merged**